### PR TITLE
[BD-46] refactor: replaced legacy checkboxes

### PR DESCRIPTION
**Pull request by PKulkoRaccoonGang**
## Description
- replaced legacy checkboxes inside DataTable and TableFilters components

### Deploy Preview

- [DataTable](https://deploy-preview-2555--paragon-openedx.netlify.app/components/datatable/)
- [TableFilters](https://deploy-preview-2555--paragon-openedx.netlify.app/components/datatable/tablefilters/)

## Merge Checklist

* [x] If your update includes visual changes, have they been reviewed by a designer? Send them a link to the Netlify deploy preview, if applicable.
* [ ] Does your change adhere to the documented [style conventions](https://github.com/openedx/paragon/blob/master/docs/decisions/0012-css-styling-conventions)?
* [x] Do any prop types have missing descriptions in the Props API tables in the documentation site (check deploy preview)?
* [x] Were your changes tested using all available themes (see theme switcher in the header of the deploy preview, under the "Settings" icon)?
* [ ] Were your changes tested in the `example` app?
* [x] Is there adequate test coverage for your changes?
* [x] Consider whether this change needs to reviewed/QA'ed for accessibility (a11y). If so, please add `wittjeff` and `adamstankiewicz` as reviewers on this PR.

## Post-merge Checklist

* [ ] Verify your changes were released to [NPM](https://www.npmjs.com/package/@edx/paragon) at the expected version.
* [ ] If you'd like, [share](https://github.com/openedx/paragon/discussions/new?category=show-and-tell) your contribution in [#show-and-tell](https://github.com/openedx/paragon/discussions/categories/show-and-tell).
* [ ] 🎉 🙌 Celebrate! Thanks for your contribution.

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @PKulkoRaccoonGang!


When this pull request is ready, tag your edX technical lead.

File: src/DataTable/filters/CheckboxFilter.jsx [hunk 1]
```diff
@@ -27,15 +26,19 @@ function CheckboxFilter({
...
-          key={headerBasedId + name}
-          checked={checkedBoxes.includes(value)}
-          onChange={() => { changeCheckbox(value); }}
-          label={<>{name} {number !== undefined && <Badge variant="light">{number}</Badge>}</>}
-        />
-      ))}
+      <Form.CheckboxSet name={Header}>
+        {filterChoices.map(({ name, number, value }) => (
+          <Form.Checkbox
+            key={headerBasedId + name}
```
[the remark this answers is not in the document]
Comment (PKulkoRaccoonGang) [reply src/DataTable/filters/CheckboxFilter.jsx]: Done

File: src/DataTable/filters/MultiSelectDropdownFilter.jsx [hunk 2]
```diff
@@ -21,20 +21,22 @@ function MultiSelectDropdownFilter({
     checkedBoxes.push(value);
     return setFilter(checkedBoxes);
   };
-  const headerBasedId = useMemo(() => `checkbox-filter-check-${getHeaderProps().key}-`, [getHeaderProps]);
+
   return (
     <DropdownButton variant="outline-primary" id={ariaLabel.current} title={Header}>
-      <div role="group" aria-label={Header} className="pgn__dropdown-filter-checkbox-group">
+      <Form.CheckboxSet className="pgn__dropdown-filter-checkbox-group" name={Header} aria-label={Header}>
```
[the remark this answers is not in the document]
Comment (PKulkoRaccoonGang) [reply src/DataTable/filters/MultiSelectDropdownFilter.jsx]: Corrected, thanks

File: src/DataTable/filters/CheckboxFilter.jsx [hunk 3]
```diff
@@ -27,15 +26,19 @@ function CheckboxFilter({
...
+      <Form.CheckboxSet name={Header}>
+        {filterChoices.map(({ name, number, value }) => (
+          <Form.Checkbox
+            key={`${headerBasedId}${name}`}
+            value={name}
+            checked={checkedBoxes.includes(value)}
+            onChange={() => changeCheckbox(value)}
+            aria-label={name}
+          >
+            {name} {number !== undefined && <Badge className="ml-2" variant="light">{number}</Badge>}
```
Comment (adamstankiewicz) [inline src/DataTable/filters/CheckboxFilter.jsx:38]: We should avoid using CSS utility classes in Paragon components as they include `!important`, which makes theming/branding difficult for consumers. Consider defining a new `.pgn__` namespaced class name or perhaps using `<Stack direction="horizontal" gap={2}>` here would make sense?

File: src/DataTable/filters/MultiSelectDropdownFilter.jsx [hunk 4]
```diff
@@ -21,20 +21,26 @@ function MultiSelectDropdownFilter({
     checkedBoxes.push(value);
     return setFilter(checkedBoxes);
   };
-  const headerBasedId = useMemo(() => `checkbox-filter-check-${getHeaderProps().key}-`, [getHeaderProps]);
+
   return (
     <DropdownButton variant="outline-primary" id={ariaLabel.current} title={Header}>
-      <div role="group" aria-label={Header} className="pgn__dropdown-filter-checkbox-group">
```
Comment (adamstankiewicz) [inline src/DataTable/filters/MultiSelectDropdownFilter.jsx:27]: [clarification] Is it intentional to remove the `role="group"` attribute here?

File: src/DataTable/filters/MultiSelectDropdownFilter.jsx [hunk 5]
```diff
@@ -21,20 +21,26 @@ function MultiSelectDropdownFilter({
...
-            id={headerBasedId}
+            value={name}
             checked={checkedBoxes.includes(value)}
-            onChange={() => { changeCheckbox(value); }}
-            label={<>{name} {number && <Badge variant="light">{number}</Badge>}</>}
-          />
+            onChange={() => changeCheckbox(value)}
+            aria-label={name}
+          >
+            {name} {number && <Badge className="ml-2" variant="light">{number}</Badge>}
```
Comment (adamstankiewicz) [inline src/DataTable/filters/MultiSelectDropdownFilter.jsx:40]: Similar comment as above regarding using CSS utility classes.

File: src/hooks/useArrowKeyNavigation.mdx [hunk 6]
```diff
@@ -64,7 +64,7 @@ the `arrowUp` and `arrowLeft` keys can be ignored for convenient editing of the
           <Form.Control type="password" placeholder="Password" />
         </Form.Group>
         <Form.Group>
-          <Form.Check type="checkbox" label="Confirm the entered data" />
+          <Form.Checkbox>Confirm the entered data</Form.Checkbox>
```
Comment (adamstankiewicz) [inline src/hooks/useArrowKeyNavigation.mdx:67]: Good catch!

File: src/DataTable/filters/CheckboxFilter.jsx (hunk 3 above)
Comment (PKulkoRaccoonGang) [reply src/DataTable/filters/CheckboxFilter.jsx:38]: Yes, you are right
I will fix it

File: src/DataTable/filters/MultiSelectDropdownFilter.jsx (hunk 4 above)
Comment (PKulkoRaccoonGang) [reply src/DataTable/filters/MultiSelectDropdownFilter.jsx:27]: Yes, Form.CheckboxSet have `role ="group"` by default

image

File: src/hooks/useArrowKeyNavigation.mdx (hunk 6 above)
Comment (PKulkoRaccoonGang) [reply src/hooks/useArrowKeyNavigation.mdx:67]: Thanks!

File: src/DataTable/filters/MultiSelectDropdownFilter.jsx (hunk 4 above)
Comment (adamstankiewicz) [reply src/DataTable/filters/MultiSelectDropdownFilter.jsx:27]: Thanks for confirming! 🎉

**Review by adamstankiewicz — APPROVED** [review]

**Comment by openedx-webhooks** [comment]
@PKulkoRaccoonGang 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/paragon&entry.867055334=https://github.com/openedx/paragon/pull/2555&entry.1484655318=https://github.com/PKulkoRaccoonGang&entry.752974735=2023-08-16+18:09&entry.1917517419=2023-09-01+12:24&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**